core/src/main/java/com/sequenceiq/cloudbreak/service/sharedservice/DatalakeService.java

```diff
@@ -0,0 +1,265 @@
+                LOGGER.debug("DatalakeResources (datalake stack id: {} ,name: {}) has found for stack: {}",
+                        datalakeResource.getDatalakeStackId(), datalakeResource.getName(), stack.getResourceCrn());
+            } else {
+                LOGGER.debug("Unable to find DatalakeResources for datalake resource id: " + stack.getDatalakeResourceId());
+            }
+        } else {
+            LOGGER.debug("Datalake resource ID was null! Going to search it by the given stack's id (id: {})", stack.getId());
+            Optional<DatalakeResources> datalakeResources = datalakeResourcesService.findByDatalakeStackId(stack.getId());
+            if (datalakeResources.isPresent()) {
+                LOGGER.debug("DatalakeResources (datalake stack id: {} ,name: {}) has found for stack: {}",
+                        datalakeResources.get().getDatalakeStackId(), datalakeResources.get().getName(), stack.getResourceCrn());
+                for (StackIdView connectedStacks : stackService.findClustersConnectedToDatalakeByDatalakeResourceId(datalakeResources.get().getId())) {
+                    AttachedClusterInfoV4Response attachedClusterInfoResponse = new AttachedClusterInfoV4Response();
+                    attachedClusterInfoResponse.setId(connectedStacks.getId());
+                    attachedClusterInfoResponse.setName(connectedStacks.getName());
+                    sharedServiceResponse.getAttachedClusters().add(attachedClusterInfoResponse);
+                }
+            } else {
+                LOGGER.debug("Unable to find DatalakeResources by the stack's id: {}", stack.getId());
+            }
+        }
+        stackResponse.setSharedService(sharedServiceResponse);
+    }
+
+    public Optional<Stack> getDatalakeStackByDatahubStack(Stack datahubStack) {
+        Long datalakeResourceId = datahubStack.getDatalakeResourceId();
+        if (datalakeResourceId != null) {
+            Optional<DatalakeResources> datalakeResource = datalakeResourcesService.findById(datalakeResourceId);
+            if (datalakeResource.isPresent() && datalakeResource.get().getDatalakeStackId() != null) {
+                Long datalakeStackId = datalakeResource.get().getDatalakeStackId();
+                Stack datalakeStack = stackService.getByIdWithListsInTransaction(datalakeStackId);
+                return Optional.of(datalakeStack);
+            }
+        }
+        LOGGER.info("There is no datalake resource id has set for the cluster.");
+        return Optional.empty();
+    }
+
+    public Optional<Stack> getDatalakeStackByStackEnvironmentCrn(Stack datahubStack) {
+        Set<DatalakeResources> datalakeResources = datalakeResourcesService
+                .findDatalakeResourcesByWorkspaceAndEnvironment(datahubStack.getWorkspace().getId(), datahubStack.getEnvironmentCrn());
+        Optional<Stack> datalakeStack = Optional.ofNullable(datalakeResources)
```

Comment:
This is only working with the previous version which generates the datalakeResourceId
But it should be the fallback solution only.
First we should try to read the new field datalakeCrn and then fallback